gslb/nodes/dq_ingestion.go

```diff
@@ -272,6 +300,202 @@ func DequeueIngestion(key string) {
+		var members []AviGSK8sObj
+		modelName := utils.ADMIN_NS + "/" + lfqdn
+		found, aviGS := agl.Get(modelName)
+		if found {
+			aviGSGraph := aviGS.(*AviGSObjectGraph)
+			// an existing GS Graph was found for the local fqdn, delete member(s) or the GS graph.
+			gslbutils.Logf("key: %s, modelName: %s, msg: %s", key, modelName,
+				"GS graph for local fqdn exists, will evaluate the members")
+			members = aviGSGraph.GetGSMembersByCluster(cname)
+			DeleteGSOrGSMembers(aviGSGraph, members, modelName, agl, sharedQ, key)
+		}
+		if len(members) == 0 {
+			// there are no members which can be added to the new GS, return
+			gslbutils.Logf("key: %s, gFqdn: %s, msg: no pending members to be added", key, gfqdn)
+			return
+		}
+
+		// a new GS object needs to be created or updated for the global fqdn
+		newModelName := utils.ADMIN_NS + "/" + gfqdn
+		found, aviGS = agl.Get(newModelName)
+		if found {
+			gslbutils.Logf("key: %s, gsName: %s, msg: GS for global FQDN already exists, will update", key,
+				aviGS.(*AviGSObjectGraph).Name)
+			aviGSGraph := aviGS.(*AviGSObjectGraph)
+			for _, m := range members {
+				aviGSGraph.AddUpdateGSMember(m)
+			}
+			agl.Save(gfqdn, aviGS.(*AviGSObjectGraph))
+			return
+		}
+
+		aviGS = NewAviGSObjectGraph()
+		aviGSGraph := aviGS.(*AviGSObjectGraph)
+		aviGSGraph.ConstructAviGSGraphFromObjects(gfqdn, members, key)
+		gslbutils.Debugf(spew.Sprintf("key: %s, gsName: %s, model: %v, msg: constructed new model", key, modelName,
+			*(aviGSGraph)))
+		agl.Save(newModelName, aviGS.(*AviGSObjectGraph))
+		PublishKeyToRestLayer(utils.ADMIN_NS, gfqdn, key, sharedQ)
+
+	case gslbutils.ObjectUpdate:
+		fqdnMap := gslbutils.GetFqdnMap()
+		// for an update operation, the fqdns have different meaning
+		prevFqdn := lfqdn
+		newFqdn := gfqdn
+		lFqdnObjs, err := fqdnMap.GetLocalFqdnsForGlobalFqdn(prevFqdn)
+		if err != nil {
+			gslbutils.Logf("key: %s, cluster: %s, prevFqdn: %s, msg: error in fetching local fqdn mapping",
+				key, cname, prevFqdn)
+		}
+		for _, f := range lFqdnObjs {
+			if f.Cluster == cname {
+				fqdnMap.DeleteFromFqdnMapping(prevFqdn, f.Fqdn, f.Cluster)
+				fqdnMap.AddUpdateToFqdnMapping(newFqdn, f.Fqdn, f.Cluster)
+			}
+		}
+
+		var members []AviGSK8sObj
+		modelName := utils.ADMIN_NS + "/" + prevFqdn
+		found, aviGS := agl.Get(modelName)
+		if found {
+			aviGSGraph := aviGS.(*AviGSObjectGraph)
+			// an existing GS Graph was found for the local fqdn, delete member(s) or the GS graph.
+			gslbutils.Logf("key: %s, modelName: %s, msg: %s", key, modelName,
+				"GS graph for local fqdn exists, will evaluate the members")
+			members = aviGSGraph.GetGSMembersByCluster(cname)
+			DeleteGSOrGSMembers(aviGSGraph, members, modelName, agl, sharedQ, key)
+		}
+		if len(members) == 0 {
+			// there are no members which can be added to the new GS, return
+			gslbutils.Logf("key: %s, gFqdn: %s, msg: no pending members to be added", key, gfqdn)
```

Comment:
I think this log message has to change. Here we are exiting because GS for prevFqdn is not found .. and I'm not sure if we should do that.

Reply:
I can add another log where the prevFqdn wasn't found.